src/components/v2/chartDeploymentHistory/chartDeploymentHistory.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Routes } from '../../../config'
+import { get, put } from '../../../services/api'
+import { ResponseType } from '../../../services/service.types'
+import { ActionResponse, ChartMetadata, DeployedAt, InstalledAppInfo } from '../../external-apps/ExternalAppService'
+
+export interface ChartDeploymentManifestDetail {
+    manifest?: string
+    valuesYaml?: string
+}
+export interface ChartDeploymentManifestDetailResponse extends ResponseType {
+    result?: ChartDeploymentManifestDetail
+}
+
+export interface DeploymentHistoryAndInstalledAppInfo {
+    deploymentHistory: ChartDeploymentDetail[]
+    installedAppInfo: InstalledAppInfo
+}
+
+export interface ChartDeploymentHistoryResponse extends ResponseType {
+    result?: DeploymentHistoryAndInstalledAppInfo
+}
+
+export interface ChartDeploymentDetail {
+    chartMetadata: ChartMetadata
+    dockerImages: string[]
+    version: number
+    installedAppVersionId?: number
+    deployedAt: DeployedAt
+}
+
+export interface RollbackReleaseRequest {
+    version: number
+    hAppId?: string
+    installedAppId?: number
+    installedAppVersionId?: number
+}
+
+interface RollbackReleaseResponse extends ResponseType {
+    result?: ActionResponse
+}
+
+export const getDeploymentHistory = (appId: string, isExternal: boolean): Promise<ChartDeploymentHistoryResponse> => {
+    const url = isExternal
+        ? `${Routes.HELM_RELEASE_DEPLOYMENT_HISTORY_API}?appId=${appId}`
```

Comment:
Here the data for the source code | values.yaml will remain the same for all of the timestamps as appId won't change for the whole application. is it the expected behavior?

Reply:
This API is giving us all deployment history details for provided appId. So, this is the expected behavior.